.github/workflows/set-metadata.yml

```diff
@@ -0,0 +1,354 @@
+          exit 0
+        fi
+
+        echo "Constructing JSON payload from raw inputs"
+        payload="{}"
+
+        # Get howso-engine-py version or run ID
+        if [[ "${{ inputs.howso-engine-py-build }}" =~ runs/([0-9]+) ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${BASH_REMATCH[1]}" '. + {"howso-engine-py": $run_id}')
+        elif [[ -n "${{ inputs.howso-engine-py-build }}" ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${{ inputs.howso-engine-py-build }}" '. + {"howso-engine-py": $run_id}')
+        fi
+
+        # Get amalgam-lang-py version or run ID
+        if [[ "${{ inputs.amalgam-lang-py-build }}" =~ runs/([0-9]+) ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${BASH_REMATCH[1]}" '. + {"amalgam-lang-py": $run_id}')
+        elif [[ -n "${{ inputs.amalgam-lang-py-build }}" ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${{ inputs.amalgam-lang-py-build }}" '. + {"amalgam-lang-py": $run_id}')
+        fi
+
+        # Get howso-validator-py version or run ID
+        if [[ "${{ inputs.howso-validator-py-build }}" =~ runs/([0-9]+) ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${BASH_REMATCH[1]}" '. + {"howso-validator-py": $run_id}')
+        elif [[ -n "${{ inputs.howso-validator-py-build }}" ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${{ inputs.howso-validator-py-build }}" '. + {"howso-validator-py": $run_id}')
+        fi
+
+        # Get howso-validator-enterprise-py version or run ID
+        if [[ "${{ inputs.howso-validator-enterprise-py-build }}" =~ runs/([0-9]+) ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${BASH_REMATCH[1]}" '. + {"howso-validator-enterprise-py": $run_id}')
+        elif [[ -n "${{ inputs.howso-validator-enterprise-py-build }}" ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${{ inputs.howso-validator-enterprise-py-build }}" '. + {"howso-validator-enterprise-py": $run_id}')
+        fi
+
+        # Get howso-synthesizer-py version or run ID
+        if [[ "${{ inputs.howso-synthesizer-py-build }}" =~ runs/([0-9]+) ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${BASH_REMATCH[1]}" '. + {"howso-synthesizer-py": $run_id}')
+        elif [[ -n "${{ inputs.howso-synthesizer-py-build }}" ]]; then
+          payload=$(echo "$payload" | jq --arg run_id "${{ inputs.howso-synthesizer-py-build }}" '. + {"howso-synthesizer-py": $run_id}')
+        fi
+
+        echo "$payload"
+        fmtd_payload=$(echo -n "$payload" | tr -d '\n')
+        echo "payload=$fmtd_payload" >> $GITHUB_OUTPUT
+
+  get-howso-synthesizer-py-details:
+    needs: ['construct-payload']
+    uses: "./.github/workflows/get-dependency-details.yml"
+    secrets: inherit
+    with:
+      owner: "howsoai"
+      repo: "howso-synthesizer-py"
+      payload: "${{ needs.construct-payload.outputs.payload }}"
+      skip-version-json-check: true
```

Comment:
Why do you have to set this to true on all these calls? Are there cases when you call this external wf without setting this to true?